Makefile

```diff
@@ -66,6 +66,7 @@ CNI_SWIFT_BUILD_DIR = $(BUILD_DIR)/cni-swift
 CNI_OVERLAY_BUILD_DIR = $(BUILD_DIR)/cni-overlay
 CNI_BAREMETAL_BUILD_DIR = $(BUILD_DIR)/cni-baremetal
 CNI_DUALSTACK_BUILD_DIR = $(BUILD_DIR)/cni-dualstack
+CNI_OVERLAY_CONSOLIDATED_BUILD_DIR = $(BUILD_DIR)/cni-overlay-consolidated
```

Comment:
I am opposed to adding a new build artifact like this, as we will then need to migrate _off_ of it in the future and back to the bare "overlay" artifact. 
Can we not simply consolidate on to "cni-overlay" immediately?

Reply:
I think that is a good point, one concern was backwards compatibility, to still have 'v4overlay' and 'dualstack' be valid options until we know 'overlay' works properly in AKS

So being able to roll back and still support 'dualstack' and 'v4overlay' was the major concern

@ashvindeodhar wdyt?

Reply:
@thatmattlong what do you think as well?

Reply:
@rbtr what do you think, about having that overlap period in AKS (where all three options are viable)?

It maybe simpler to consolidate immediately, we would just need to keep these changes right in sync with AKS-RP changes and a new CNS version, might miss something

Reply:
I was specifically referring to the tgz build artifacts, but looking further I think maybe too many things are changing in this single PR (or they should be reordered).
Should the changes actually be:
- release CNI that supports the consolidated `"executionMode":"swift"` and ipam mode `overlay`
  - this should be end-to-end and should collapse v4/dualstack overlay in to the single `cni-overlay` build artifact
- release dropgz with ^
- release CNS that supports the consolidated `overlay` conflist generation
- migrate AKS to consolidate `overlay` config argument everywhere
- deprecate and remove v4overlay/dualstack overlay everywhere

vs the updates you have proposed, this way we should never need to update anything to an intermediate "overlay-consolidated" that we later need to change back.

Reply:
Oh I didn't know about `"executionMode": "swift"` conslidation, is there someone working on that currently?

> release dropgz with ^

After releasing dropgz, would we update AKS-RP anywhere?

> release CNS that supports the consolidated overlay conflist generation

Or do we wait till after releasing CNS to make any AKS-RP changes?

Reply:
We update azure-cni binary and conflist at once, by virtue of tarball
	> Check this new binary and conflist don't break V4
        > Maybe add pipeline step for overlay cni
       > what aks checks are there?
       > Still would release dualstack version for backcompat

Relesae dropgz with ^
Rollout this everywhere, ~4 weeks (CNI is everywhere, and supports 'overlay' mode)
       Can make this faster by cutting new Agent baker image for dropgz right after tarball is release in our repo
Add a third 'overlay' generator, until cns image is rolled out (4 weeks)
Change the configmap, to use 'overlay'
Remove 'dualstack' and 'v4overlay' in CNS and CNI

Unknowns:
executionMode v4swift, what is it? Do we need it in v4overlay?